ecs/example/bin/ssh-to-host

```diff
@@ -1,30 +1,51 @@
-#!/bin/sh
-
-# Connects to the host created by the example using SSH proxied through a bastion node.
-# Requires jq to be installed for parsing terraform output -json
-# https://stedolan.github.io/jq/
-
+#!/usr/bin/env bash
+
+# Connects to the host created by the example using SSH proxied through SSM.
+#
+# Requires:
+# - aws cli https://docs.aws.amazon.com/cli/latest/userguide/install-cliv1.html
+# - aws cli session manager plugin https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html
+# - openssh
+#
+# Based on https://github.com/elpy1/ssh-over-ssm/blob/master/ssh-ssm.sh
+#
 # Usage:
 # bin/ssh-to-host
 
-set -e
-
-echo >&2 "Fetching outputs..."
-output=$(terraform output -no-color -json)
+set -euo pipefail
 
-host=$(printf '%s' "$output" | jq -r '.host_ip.value')
-host_user=ec2-user
-
-bastion_host=$(printf '%s' "$output" | jq -r '.bastion.value.host')
-bastion_private_key=$(printf '%s' "$output" | jq -r '.bastion.value.private_key')
-bastion_user=ec2-user
+ssh_key_path=$(mktemp)
+ssh_pub_key_path="$ssh_key_path.pub"
 
-key_path=$(mktemp)
-trap "rm -f \"$key_path\"" EXIT
-chmod 0600 "$key_path"
-echo "$bastion_private_key" > "$key_path"
+function cleanup {
+  echo >&2 "Removing key $ssh_key_path..."
+  rm -f "$ssh_key_path" "$ssh_pub_key_path"
+}
+trap cleanup EXIT
 
-bastion_proxy="ssh -F /dev/null -W %h:%p -i \"$key_path\" $bastion_user@$bastion_host"
+echo >&2 "Generating key $ssh_key_path..."
+rm "$ssh_key_path"
+ssh-keygen -q -t ed25519 -N '' -f "$ssh_key_path" -C "temporary key"
```

Comment:
What do you think about adding any information identifying the requester as a comment? I do not mean to improve security but rather have some logging information in case, for example, a key is not removed automatically for example for one person - maybe IP + timestamp?

Reply:
It might be possible to get the AWS user name from AWS CLI, so it might be possible to do something similar to how systems manager is identifying sessions. Will investigate

Reply:
Since this is just an example script maybe this should be moved to our devstarter.

Reply:
Added some additional metadata:
- user name - taken from `whoami`
- user external ip
- current date in ISO8601 format